src/openforms/formio/api/views.py

```diff
@@ -59,13 +63,19 @@ def post(self, request, *args, **kwargs):
                 content_type="text/plain",
             )
 
+        submission = serializer.validated_data["submission"]
+
+        if not owns_submission(request, submission.uuid):
+            raise PermissionDenied()
```

Comment:
missing tests for permission check, but alternatively you can also make the queryset for the `submission` field in the serializer dynamic based on `self.context["request"].session` to filter out only submissions that are owned.

I find permission checks outside of permission classes not the most elegant pattern.